core/app/models/concerns/spree/user_address.rb

```diff
@@ -40,9 +40,11 @@ def address_not_associated_with_other_user
         errors.add(:ship_address_id, :belongs_to_other_user) if ship_address&.user_id && self != ship_address&.user
       end
 
-      def address_not_associated_with_placed_order
-        errors.add(:bill_address_id, :associated_with_completed_order) if orders.complete.where(bill_address: bill_address_id).any?
-        errors.add(:ship_address_id, :associated_with_completed_order) if orders.complete.where(ship_address: ship_address_id).any?
+      def address_not_deprecated_in_completed_order
+        errors.add(:bill_address_id, :associated_with_completed_order) if
+          orders.complete.joins(:bill_address).where.not(spree_addresses: { deleted_at: nil }).where(bill_address: bill_address_id).any?
```

Comment:
I think it'd be worth putting this bit: `where.not(spree_addresses: { deleted_at: nil })` into a scope.

Reply:
@KacperMekarski please use `not_deleted` scope